app/src/main/java/org/stepic/droid/code/ui/CodeAnalyzer.kt

```diff
@@ -0,0 +1,104 @@
+package org.stepic.droid.code.ui
+
+import org.stepic.droid.util.countWhile
+import org.stepic.droid.util.substringOrNull
+
+/**
+ * Class for smart code analyzing
+ */
+object CodeAnalyzer {
```

Comment:
It is good idea to change it to class and provide with CodeScope for lazy initialization of fields. But we can make it in future